File: PlatformAPI/Models/Quizzes/Question.cs
using System.ComponentModel.DataAnnotations;

namespace PlatformAPI.Models.Quizzes
{
    public class Question
    {
        public int Id { get; set; }

        public string Description { get; set; }

        public int QuestionTypeId { get; set; }

        public QuestionType QuestionType { get; set; }

        public int SortOrder { get; set; }

        public bool IsActive { get; set; }

        public bool IsPublished { get; set; }

        public QuizQuestion QuizQuestion { get; set; }
    }
}

File: FrontEnd/src/types/Questions/Question.ts
import type { AnswerChoice, TakerAnswerChoice } from "../AnswerChoices/AnswerChoice";
import { QuestionType } from "../Questions/QuestionType";



// export interface Question {
//     id: number;
//     description: string;
//     sortOrder: number;
//     answerChoices?: AnswerChoice[];
//     isPublished: boolean;
// }

export interface QuestionBase {
    id: number;
    description: string;
    sortOrder: number;
    isPublished: boolean;
}

export interface Question extends QuestionBase {
    answerChoices?: AnswerChoice[];
}

export interface TakerQuestion extends QuestionBase {
    answerChoices?: TakerAnswerChoice[];
    questionType: QuestionType;
}